trieste/models/gpflow/models.py

```diff
@@ -211,6 +232,183 @@ def evaluate_loss_of_model_parameters() -> tf.Tensor:
 
         multiple_assign(self.model, current_best_parameters)
 
+    def conditional_predict_f(
+        self, query_points: TensorType, additional_data: Dataset
+    ) -> tuple[TensorType, TensorType]:
+        """
+        Returns the marginal GP distribution at query_points conditioned on both the model
+        and some additional data, using exact formula.
+
+        :param query_points: Set of query points with shape [M, D]
+        :param additional_data: Dataset with query_points with shape [..., N, D] and observations
+                 with shape [..., N, L]
+        :return: mean_new: predictive variance at query_points, with shape [..., M, L],
+                 and var_new: predictive variance at query_points, with shape [..., M, L]
+        """
+
+        tf.debugging.assert_shapes(
+            [
+                (additional_data.query_points, [..., "N", "D"]),
+                (additional_data.observations, [..., "N", "L"]),
+                (query_points, ["M", "D"]),
+            ],
+            message="additional_data must have query_points with shape [..., N, D]"
+            " and observations with shape [..., N, L], and query_points "
+            "should have shape [M, D]",
+        )
+
+        if isinstance(self.model, SGPR):
+            raise NotImplementedError("Conditional predict f is not supported for SGPR.")
+
+        mean_old, cov_old = self.model.predict_f(
+            additional_data.query_points, full_cov=True
+        )  # [..., N, L], [..., L, N, N]
+        mean_new, var_new = self.model.predict_f(query_points, full_cov=False)  # [M, L], [M, L]
+
+        cov_cross = self.covariance_between_points(
+            additional_data.query_points, query_points
+        )  # [..., L, N, M]
+
+        cov_shape = tf.shape(cov_old)
+        noise = self.model.likelihood.variance * tf.eye(
+            cov_shape[-2], batch_shape=cov_shape[:-2], dtype=cov_old.dtype
+        )
+        L_old = tf.linalg.cholesky(cov_old + noise)  # [..., L, N, N]
+        A = tf.linalg.triangular_solve(L_old, cov_cross, lower=True)  # [..., L, N, M]
+        var_new = var_new - leading_transpose(
+            tf.reduce_sum(A ** 2, axis=-2), [..., -1, -2]
+        )  # [M, L]
+
+        mean_old_diff = additional_data.observations - mean_old  # [..., N, L]
```

Comment:
how `additional_data.observations` makes sense here for the batch case where we don't really have these observations?

Reply:
Observations are only useful for the mean. If you look at the tIMSE PR, you can see that I used dummy values (zeros) because tIMSE only uses the variance. In the kriging believer PR, we need those values in general, you need to choose the way you emulate them.

Reply:
would it be a significant enough saving if we would have a method for updating the variances alone?

Reply:
it would indeed save one triangular solve and one matmul. I could add a Boolean with skip_mean_update=False, but is it worth the extra code?